src/number_theory/arithmetic_function.lean

```diff
@@ -311,18 +344,18 @@ lemma pmul_comm [comm_monoid_with_zero R] (f g : arithmetic_function R) :
   f.pmul g = g.pmul f :=
 by { ext, simp [mul_comm] }
 
-variable [monoid_with_zero R]
```

Comment:
Why is this now a stronger constraint (making some of the lemmas below weaker)?

Reply:
Because I changed the way `zeta` works to use `nat_cast`. In principle, `monoid_with_zero` might still work for `pmul_zeta`, but at the moment, it causes a max-class-inference error.